examples/differential_correction_coupled.py

```diff
@@ -0,0 +1,237 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Thu Oct 25 17:19:30 2019
+
+@author: Wenyang Lyu and Shibabrat Naik
+
+Script to compute unstable periodic orbits at specified energies for the 
+coupled quartic Hamiltonian using differential correction
+"""
+
+import numpy as np
+from scipy.integrate import solve_ivp
+import time
+
+import matplotlib as mpl
+from matplotlib import cm
+from mpl_toolkits.mplot3d import Axes3D
+import matplotlib.pyplot as plt
+mpl.rcParams['mathtext.fontset'] = 'cm'
+mpl.rcParams['mathtext.rm'] = 'serif'
+
+# This needs testing for installation 
+import sys
+sys.path.insert(0, './src/')
+import differential_correction as diffcorr
+import coupled_quartic_hamiltonian as coupled
+# This needs testing for installation 
+
+#%% Setting up parameters for the method and the Hamiltonian system
+
+save_final_plot = True
+show_final_plot = False
+N = 4               # dimension of phase space
+omega=1.00
+EPSILON_S = 0.0     # Energy of the saddle
+alpha = 1.00
+beta = 1.00
+epsilon= 1e-1
+parameters = np.array([1, omega, EPSILON_S, alpha, beta, omega, epsilon])
+eqNum = 1
+#deltaE_vals = [0.01, 0.1, 1.00, 2.0, 4.0]
+#linecolor = ['b','r','g','m','c']
+deltaE_vals = [0.1, 1.00]
+linecolor = ['b','r']
+
+
+eqPt = diffcorr.get_eq_pts(eqNum, coupled.init_guess_eqpt_coupled, \
+                            coupled.grad_pot_coupled, parameters)
+
+#energy of the saddle eq pt
+eSaddle = diffcorr.get_total_energy([eqPt[0],eqPt[1],0,0], \
+                                    coupled.pot_energy_coupled, parameters) 
+#If orbit is an n-array, e.g. orbit = [orbit_0, orbit_1, ..., orbit_n]
+# 
+#e = np.zeros(n,1)    
+#for i in range(n):
+#    e[i] = get_total_energy_deleonberne(orbit[i], parameters)
+#e = np.mean(e)
+
+#%%
+nFam = 100 # use nFam = 10 for low energy
+
+# first two amplitudes for continuation procedure to get p.o. family
+Ax1  = 2.e-5 # initial amplitude (1 of 2) values to use: 2.e-3
+Ax2  = 2*Ax1 # initial amplitude (2 of 2)
+
+t = time.time()
+
+#  get the initial conditions and periods for a family of periodic orbits
+po_fam_file = open("x0_diffcorr_fam_eqPt%s_coupled.dat" %eqNum,'a+')
```

Comment:
Use a `with open` here (https://docs.python.org/3/tutorial/inputoutput.html#reading-and-writing-files)

Reply:
Done.